File: projects/proj01/RectangleArea.java


/* RectangleArea.java computes the area of a rectangle.
 *
 * Input: The width and height of the rectangle.
 * Precondition: The inputs are positive numbers.
 * Output: The area of the rectangle.
 *
 * Completed by: Aaron Santucci ajs94
 * Date: February 6, 2017
 **********************************************************/


import java.util.Scanner;  // include class for Scanner

public class RectangleArea {

     /* function circleArea() computes a circle's area, given its radius.
      * Parameter: w, a double
      * 	   h, a double
      * Precondition: w and h is not negative.
      * Returns: the area of the rectangle
      */
     public static double rectangleArea(double w, double h) {
        return w * h;
     }
	
  // main program
  public static void main(String[] args) {
      // prompt for inputs width and height
      System.out.println("To compute the area of a rectangle,");

	// get the width
      System.out.print(" enter its height: ");
      // Create a connection named keyboard to standard in
      Scanner keyboard = new Scanner(System.in); 
      //Get the number from the user
      double width = keyboard.nextDouble();

	// get the height
      System.out.print(" enter its width: ");
      //Get the number from the user
      double height = keyboard.nextDouble();

      // output area
      System.out.println("The area is " + rectangleArea(width, height)); 
  } // main method
} // class RectangleArea

File: projects/proj01/rectangle_area.rb
#! /usr/bin/ruby
# rectangle_area.rb computes the area of a rectangle given its width and height
# 
# Input: the width of the rectangle, 
#        the height of the rectangle.
# Precondition: the width and height are not negative
# Output: the area of the rectangle.
#
# Completed by: Aaron Santucci ajs94
# Date: February
###############################################################

# function rectangleArea returns a rectangle's area, given its width and height
# Parameters: r, a number
# Precondition: w, h > 0.
# Returns: the area of a rectangle whose width is w and heigh h.
def rectangleArea(w, h)
    w * h 
end

if __FILE__ == $0
   puts "To compute the area of a rectangle,"
   print " enter its width: "
   width = gets.chomp.to_f
   print " enter its height: "
   height = gets.chomp.to_f
   print "The rectangle's area is: "
   puts rectangleArea(width, height)
end